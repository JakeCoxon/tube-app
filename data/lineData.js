
const lines = [
  {
    id: 'bakerloo',
    apiKey: 'bakerloo',
    name: 'Bakerloo',
    color: '#996633',
    backgroundColor: '#996633',
  },
  {
    id: 'central',
    apiKey: 'central',
    name: 'Central',
    color: '#CC3333',
    backgroundColor: '#CC3333',
  },
  {
    id: 'circle',
    apiKey: 'circle',
    name: 'Circle',
    color: '#FFCC00',
    backgroundColor: '#E9B900', // Circle line needs higher contrast for white fonts

  },
  {
    id: 'district',
    apiKey: 'district',
    name: 'District',
    color: '#006633',
    backgroundColor: '#006633',
  },
  {
    id: 'hsmith-city',
    apiKey: 'hammersmith-city',
    name: 'Hammersmith & City',
    color: '#CC9999',
    backgroundColor: '#CC9999',
  },
  {
    id: 'jubilee',
    apiKey: 'jubilee',
    name: 'Jubilee',
    color: '#868F98',
    backgroundColor: '#868F98',
  },
  {
    id: 'metropolitan',
    apiKey: 'metropolitan',
    name: 'Metropolitan',
    color: '#660066',
    backgroundColor: '#660066',
  },
  {
    id: 'northern',
    apiKey: 'northern',
    name: 'Nothern',
    color: '#000000',
    backgroundColor: '#000000',
  },
  {
    id: 'piccadilly',
    apiKey: 'piccadilly',
    name: 'Piccadilly',
    color: '#0019a8',
    backgroundColor: '#0019a8',
  },
  {
    id: 'victoria',
    apiKey: 'victoria',
    name: 'Victoria',
    color: '#0099CC',
    backgroundColor: '#0099CC',
  },
  {
    id: 'waterloo-city',
    apiKey: 'waterloo-city',
    name: 'Waterloo & City',
    color: '#66CCCC',
    backgroundColor: '#66CCCC',
  },
  
  // Others
  {
    id: 'dlr',
    apiKey: 'dlr',
    name: 'DLR',
    color: '#009999',
    backgroundColor: '#009999',
  },
  
  {
    id: 'overground',
    apiKey: 'london-overground',
    name: 'Overground',
    color: '#EF7B10',
    backgroundColor: '#EF7B10',
  },
  

]

export default lines;