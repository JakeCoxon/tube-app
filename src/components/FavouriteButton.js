import React from 'react';
import { StyleSheet, Text, View, TouchableHighlight, ScrollView, ActivityIndicator, Alert } from 'react-native';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import _ from 'lodash'

import { MaterialCommunityIcons } from '@expo/vector-icons';

import { Row, Spacing, Margin, Padding, Layer } from 'tube-app/src/components/presentation.js'

import { TOGGLE_FAVOURITE } from 'tube-app/src/favourite/favouriteActions.js'

const lineIdFromProps = (props) => {
  const { lineId } = props.navigation.state.params || {}
  return lineId;
}

const mapStateToProps = (state, props) => ({
  isFavourite: state.favourite == lineIdFromProps(props),
})

const mapDispatchToProps = (dispatch, props) => ({
  onPressToggle: (isCurrentlyFavourite) => {
    const toggle = () => dispatch({ type: TOGGLE_FAVOURITE, payload: lineIdFromProps(props) })

    Alert.alert(
      isCurrentlyFavourite ? "This line is currently favourited" : "This line is not currently favourited",
      '',
      [
        isCurrentlyFavourite ? { text: 'Unfavourite this line', onPress: toggle }
        : { text: 'Favourite this line', onPress: toggle },
        { text: 'Cancel', onPress: () => {}, style: 'cancel' },
      ],
    )
    
  }
})

function FavouriteButton({ isFavourite, onPressToggle }) {
  return <TouchableHighlight 
          onPress={() => onPressToggle(isFavourite)}
          underlayColor='rgba(0,0,0,0.3)'>
        <Padding small>
          <MaterialCommunityIcons name={isFavourite ? 'heart' : 'heart-outline'} color='white' size={24} />
        </Padding>
      </TouchableHighlight>
}

export default connect(mapStateToProps, mapDispatchToProps)(FavouriteButton)

â€Š