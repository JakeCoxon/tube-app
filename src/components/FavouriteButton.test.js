import React from 'react'
import renderer from 'react-test-renderer'
import PropTypes from 'prop-types'

import FavouriteButton from 'tube-app/src/components/FavouriteButton.js'
import { createStore } from 'redux'
import statusData from 'tube-app/data/testData1.js'
import lineData from 'tube-app/data/lineData.js'
import TestProvider from 'tube-app/test/TestProvider.js'

const render = (props) => 
  <TestProvider {...props}>
    <FavouriteButton {...props} />
  </TestProvider>

it('renders unfavourited', () => {

  const props = {
    navigationParams: {
      lineId: 'dlr'
    },
    storeState: {
      favourite: undefined
    }
  }

  const tree = renderer
    .create(render(props))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders favourited', () => {

  const props = {
    navigationParams: {
      lineId: 'dlr'
    },
    storeState: {
      favourite: 'dlr'
    }
  }

  const tree = renderer
    .create(render(props))
    .toJSON();
  expect(tree).toMatchSnapshot();
});



