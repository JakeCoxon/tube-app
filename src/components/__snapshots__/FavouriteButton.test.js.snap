// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders favourited 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 8,
        },
        undefined,
        Object {
          "paddingRight": 8,
        },
        undefined,
        Object {
          "paddingTop": 8,
        },
        undefined,
        Object {
          "paddingBottom": 8,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        null,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;

exports[`renders unfavourited 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 8,
        },
        undefined,
        Object {
          "paddingRight": 8,
        },
        undefined,
        Object {
          "paddingTop": 8,
        },
        undefined,
        Object {
          "paddingBottom": 8,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        null,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
